src/System.Data.Common/src/System/Data/DataTable.cs

```diff
@@ -372,8 +372,7 @@ internal void SerializeTableSchema(SerializationInfo info, StreamingContext cont
                 info.AddValue(string.Format(formatProvider, "DataTable.DataColumn_{0}.DefaultValue", i), Columns[i].DefaultValue);
                 info.AddValue(string.Format(formatProvider, "DataTable.DataColumn_{0}.ReadOnly", i), Columns[i].ReadOnly);
                 info.AddValue(string.Format(formatProvider, "DataTable.DataColumn_{0}.MaxLength", i), Columns[i].MaxLength);
-                info.AddValue(string.Format(formatProvider, "DataTable.DataColumn_{0}.DataType", i), Columns[i].DataType);
-
+                info.AddValue(string.Format(formatProvider, "DataTable.DataColumn_{0}.DataType", i), Columns[i].DataType.AssemblyQualifiedName);
```

Comment:
When someone tries to deserialize on .NET Core a DataTable serialized on NETFX using SerializationFormat.Binary, it will throw (because Type is not serializable). Is the error reasonable and if not can we improve it?

Reply:
if I understand you correctly, you are talking about the situation when DataTable is serialized on netfx and when we are trying to deserialize it in .net core, it will throw an exception as ```DataTable.DataColumn_{0}.DataType``` will be of type ```Type``` rather than string.
We can add a private optional field whether the data has been serialized on netfx or on netcore. Then we can take appropriate action while deserializing

Reply:
I don't think you need a field. In .NET Framework it reads off the value here:

`dc.SimpleType = (SimpleType)info.GetValue(string.Format(formatProvider, "DataTable.DataColumn_{0}.SimpleType", i), typeof(SimpleType));`

In .NET Core you would check whether `info` contained a value for `DataTable.DataColumn_{0}.SimpleType` and if so then it came from .NET Framework and you can't deal with that so you have to bail with a SerializationException with a nice message explaining that .NET Core cannot deserialize DataTables sewrialized on .NET Framework using SerializationFormat.Binary and they should use SerializationFormat.Xml or some other serialization mechanism.

Reply:
There is no way to check whether info contains ```string.Format(formatProvider, "DataTable.DataColumn_{0}.SimpleType", i), typeof(SimpleType)``` this or not

Reply:
I believe throwing a SerializationException is enough here. We don't do anything fancy at other places where can't deserialize unsupported types.

Reply:
Is there any reason we are not changing the identifier (`DataTable.DataColumn_{0}.DataType`)? We are now storing a different "kind" of data inside it, although for the same purpose. It seems to me it may be clearer to rename it.

Even if later we modified .NET Framework to read/write this "new" format, it does not require us use the same name.

Reply:
yeah it would be good to change it . I suggest```DataTable.DataColumn_{0}.DataType_AssemblyQualifiedName``` ?

Reply:
Perfect

Reply:
@ViktorHofer 

> You can't browse through the SerializationInfo object (by using its internal helper methods) outside of System.Private.CoreLib. 

I would have expected GetValueNoThrow to do the job? But it's fine as is.